artemis-server/src/main/java/org/apache/activemq/artemis/core/config/impl/ConfigurationImpl.java

```diff
@@ -773,6 +774,17 @@ public <T> T convert(Class<T> type, Object value) {
          }
       }, TransformerConfiguration.class);
 
+      beanUtils.getConvertUtils().register(new Converter() {
+         @Override
+         public <T> T convert(Class<T> type, Object value) {
+            //we only care about DATABASE type as it is the only one used
+            if (StoreConfiguration.StoreType.DATABASE.toString().equals(value)) {
+               return (T) new DatabaseStorageConfiguration();
+            }
+            return (T) null;
```

Comment:
I think throw some runtimeexception here - value must be DATABASE, otherwise the error on the next set will be hard to debug

Reply:
Ive changed this